konstructio/gitops-template
# Build real clusters in Azure GitHub

## Description

Create physical clusters for Azure GitHub

## Related Issue(s)

Fixes #

## How to test

Build an Azure cluster pointing to this branch in the GitOps templates. Until the console work is done, this will require manually adding the `gitops` repo:
1. Copy `templates/workload-cluster` to `registry/<cluster-name>`
2. Move `registry/<cluster-name>/appproject-workload-cluster.yaml` to `registry/clusters/<mgmt>/components/argocd-appprojects`
3. Move `registry/<cluster-name>/registry-workload-cluster.yaml` to `registry/clusters/<mgmt>`
4. Search and replace the following values:
   - `<WORKLOAD_CLUSTER_NAME>`: `<cluster-name>`
   - `<WORKLOAD_CLUSTER_REGION>`: `ukwest`
   - `<WORKLOAD_NODE_COUNT>`: `3`
   - `<WORKLOAD_NODE_TYPE>`: `Standard_D2s_v4`
   - `<WORKLOAD_EXTERNAL_DNS_DOMAIN_NAME>`: `<domain name>` (see `/registry/clusters/<mgmt>/components/external-dns/application.yaml`)
